matching-core/src/main/java/com/graphhopper/matching/LocationIndexMatch.java

```diff
@@ -1,153 +0,0 @@
-/*
- *  Licensed to GraphHopper GmbH under one or more contributor
- *  license agreements. See the NOTICE file distributed with this work for 
- *  additional information regarding copyright ownership.
- * 
- *  GraphHopper GmbH licenses this file to you under the Apache License, 
- *  Version 2.0 (the "License"); you may not use this file except in 
- *  compliance with the License. You may obtain a copy of the License at
- * 
- *       http://www.apache.org/licenses/LICENSE-2.0
- * 
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
- */
-package com.graphhopper.matching;
-
-import com.carrotsearch.hppc.procedures.IntProcedure;
-import com.graphhopper.coll.GHBitSet;
-import com.graphhopper.coll.GHIntHashSet;
-import com.graphhopper.coll.GHTBitSet;
-import com.graphhopper.routing.util.EdgeFilter;
-import com.graphhopper.storage.GraphHopperStorage;
-import com.graphhopper.storage.index.LocationIndexTree;
-import com.graphhopper.storage.index.QueryResult;
-import com.graphhopper.util.EdgeExplorer;
-import com.graphhopper.util.EdgeIteratorState;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.List;
-
-/**
- *
- * @author Peter Karich
- */
-public class LocationIndexMatch extends LocationIndexTree {
```

Comment:
I think it makes sense for the method `findCandidateLocations` to be part of `HmmTimeStep` as it's currently laid out - it felt cleaner to me (mainly because the code is only used for this one purpose). However, if Graphhopper had this index logic implemented, then yes, it'd be nicer, and the `findCandidateLocations` method would be much simpler. I'm not too worried though - if you want me to change it, I'm happy to.